src/main/java/frc/team2412/robot/commands/indexer/IndexBitmapCommand.java

```diff
@@ -102,87 +104,91 @@ public IndexBitmapCommand(final IndexerSubsystemSuperStructure indexerMotorSubsy
-			boolean runLift = ((frontIndexDirection == IndexDirection.IN)
-					|| (frontIndexDirection == IndexDirection.OUT))
-					|| ((backIndexDirection == IndexDirection.IN) || (backIndexDirection == IndexDirection.OUT));
-
-			if (runLift) {
-				lastIndexRunTimeMicroSec = RobotController.getFPGATime();
-			} else if (RobotController.getFPGATime() < lastIndexRunTimeMicroSec
-					+ 1000 * IndexerConstants.LIFT_DOWN_MS_DURATION_AFTER_SHUFFLE) {
-				runLift = true;
-			}
+	private void runCommandPresent(IndexCommandEntry indexCommand, IntakeDirection intakeDirection,
+			final boolean intakeOn) {
+		final IndexDirection frontIndexDirection = indexCommand.getFrontIndexDirection(intakeOn);
+		final IndexDirection backIndexDirection = indexCommand.getBackIndexDirection(intakeOn);
+		boolean runLift = shouldRunLift(frontIndexDirection, backIndexDirection);
+
+		if (runLift) {
+			lastIndexRunTimeMicroSec = RobotController.getFPGATime();
+		} else if (RobotController.getFPGATime() < lastIndexRunTimeMicroSec
+				+ 1000 * IndexerConstants.LIFT_DOWN_MS_DURATION_AFTER_SHUFFLE) {
+			runLift = true;
+		}
 
-			m_indexerSubsystem.getIndexerMotorLiftSubsystem()
-					.set(runLift ? IndexerConstants.LIFT_DOWN_SPEED_FOR_INDEX : 0.0);
-			switch (intakeDirection) {
+		m_indexerSubsystem.getIndexerMotorLiftSubsystem()
+				.set(runLift ? IndexerConstants.LIFT_DOWN_SPEED_FOR_INDEX : 0.0);
+		switch (intakeDirection) {
 			case BOTH:
 			case FRONT:
-				m_indexerSubsystem.getIndexerMotorFrontSubsystem().set(getIndexerMotorSpeed(frontIndexDirection));
-				m_indexerSubsystem.getIndexerMotorBackSubsystem().set(getIndexerMotorSpeed(backIndexDirection));
+				m_indexerSubsystem.setFrontAndBack(getIndexerMotorSpeed(frontIndexDirection),
+						getIndexerMotorSpeed(backIndexDirection));
 				break;
 			case BACK:
 				// Swap the front & back motor values since the IndexCommandEntry assumes intake
 				// from the front
-				m_indexerSubsystem.getIndexerMotorFrontSubsystem().set(getIndexerMotorSpeed(backIndexDirection));
-				m_indexerSubsystem.getIndexerMotorBackSubsystem().set(getIndexerMotorSpeed(frontIndexDirection));
+				m_indexerSubsystem.setFrontAndBack(getIndexerMotorSpeed(backIndexDirection),
+						getIndexerMotorSpeed(frontIndexDirection));
 				break;
 			default:
-				m_indexerSubsystem.getIndexerMotorFrontSubsystem().set(0);
-				m_indexerSubsystem.getIndexerMotorBackSubsystem().set(0);
+				m_indexerSubsystem.setFrontAndBack(0, 0);
 				break;
-			}
-		} else {
-			m_indexerSubsystem.getIndexerMotorFrontSubsystem().set(0);
-			m_indexerSubsystem.getIndexerMotorBackSubsystem().set(0);
-			m_indexerSubsystem.getIndexerMotorLiftSubsystem().set(0);
 		}
 	}
 
+	private boolean shouldRunLift(final IndexDirection frontIndexDirection, final IndexDirection backIndexDirection) {
+		return ((frontIndexDirection == IndexDirection.IN) || (frontIndexDirection == IndexDirection.OUT))
```

Comment:
Why are there extra parentheses here separating the front and back, isn't this if any are true then run lift? Not a code thing, just wondering if there is a reason

Reply:
idk i didnt mess with it. it just makes it clearer I guess

Reply:
Yeah this just runs the lift if any are true, so the parentheses aren't needed. I think they were originally included to separate the front index logic from the back index's.